json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php

```diff
@@ -233,3 +239,151 @@ protected function result() {
 		);
 	}
 }
+
+class Jetpack_JSON_API_Sync_Checkout_Endpoint extends Jetpack_JSON_API_Sync_Endpoint {
+	protected function result() {
+		$args = $this->input();
+		$queue_name = $this->validate_queue( $args['queue'] );
+
+		if ( is_wp_error( $queue_name ) ){
+			return $queue_name;
+		}
+
+		if ( is_int( $args[ 'number_of_items' ] ) && (int) $args[ 'number_of_items' ] < 1 && (int) $args[ 'number_of_items' ] > 100 ) {
+			return new WP_Error( 'invalid_number_of_items', 'Number of items needs to be an integer that is larger than 0 and less then 100', 400 );
+		}
+
+		require_once JETPACK__PLUGIN_DIR . 'sync/class.jetpack-sync-queue.php';
+		$queue = new Jetpack_Sync_Queue( $queue_name );
+
+		if ( $queue->size() < 1 ) {
+			return new WP_Error( 'queue_size', 'The queue is empty and there is nothing to send', 400 );
+		}
+
+		$encode = ( isset( $args['encode'] ) && $args['encode'] ? true : false );
+		$codec_name = null;
+		if ( $encode ) {
+			require_once JETPACK__PLUGIN_DIR . 'sync/class.jetpack-sync-json-deflate-array-codec.php';
```

Comment:
rather than all of this, just call:

```
Jetpack_Sync_Sender::get_instance()->get_codec()->name();
```

That will reduce the maintenance burden here as the default codecs change over time.